masonry/src/tree_arena.rs

```diff
@@ -252,14 +271,47 @@ impl<'a, Item> ArenaRefChildren<'a, Item> {
         self.children
             .iter()
             .find(|child| child.id == id)
-            .map(|child| child.arena_ref(self.id))
+            .map(|child| child.arena_ref(self.id, self.parents_map.parents_map))
     }
 
     // TODO - This method could not be implemented with an actual arena design.
     // It's currently used for some sanity-checking of widget code, but will
     // likely be removed.
     pub(crate) fn iter_children(&self) -> impl Iterator<Item = ArenaRef<'_, Item>> {
-        self.children.iter().map(|child| child.arena_ref(self.id))
+        self.children
+            .iter()
+            .map(|child| child.arena_ref(self.id, self.parents_map.parents_map))
+    }
+
+    /// Find an item in the tree.
+    ///
+    /// Returns a shared reference to the item.
+    ///
+    /// ## Complexity
+    ///
+    /// O(Depth). In future implementations, this will be O(1).
+    pub fn find(self, id: u64) -> Option<ArenaRef<'a, Item>> {
```

Comment:
Why does this consume `self`? `ArenaRefChildren` doesn't seem to be `Copy`

Reply:
It does. You can grep for `Copy for ArenaRefChildren` to find it.

It's not a derive, because derive creates an implementation conditional on `Item: Copy`, which isn't what we want.

Reply:
I was reviewing this in GitHub's UI, so that wasn't visible.

My expectation was that such a fundamental impl would not be hiding somewhere else...